### ISSUE TITLE
Override COMMIT on SQL table create from default (*CHG) to *NONE not processed

### ISSUE THREAD

**Issue by sbestgen** [IBM/ibmi-bob#121]
Trying to change the default of COMMIT(*CHG) to COMMIT(*NONE) on an SQL create table RESULTS- not working:  

SQLs := BATCHPF.FILE POOLLOG.FILE RSPLOG.FILE SYSLOG.FILE RESULTS.FILE 

POOLLOG.FILE: $(d)/POOLLOG.TABLE
RSPLOG.FILE: $(d)/RSPLOG.TABLE
SYSLOG.FILE: $(d)/SYSLOG.TABLE
BATCHPF.FILE: $(d)/BATCHPF.TABLE
RESULTS.FILE: private COMMIT := *NONE
RESULTS.FILE: $(d)/RESULTS.TABLE

> make -k BUILDVARSMKPATH="/tmp/tmpaxo4_y6q" -k BOB="/QOpenSys/pkgs/lib/bob" -f "/QOpenSys/pkgs/lib/bob/mk/Makefile" all
=== Creating SQL TABLE from Sql statement [RESULTS.TABLE]
RUNSQLSTM srcstmf('/home/SBESTGEN/BATCHWL/QSQLSRC/RESULTS.TABLE') DBGVIEW(*SOURCE) TGTRLS() OUTPUT(*PRINT)
✕ Failed to create RESULTS.FILE!

Objects:             1 failed 0 succeed 1 total
 > Failed objects:    RESULTS.FILE

**Comment by edmundreinhardt** [comment]
Hi Susan, what was the contents of RESULTS.TABLE stream file?

**Comment by sbestgen** [comment]
The RESULTS stream files does a CREATE TABLE AS, built over the RSPLOG table.  I’ve copied the source for both stream files below.

***@***.***

***@***.***
From: Edmund Reinhardt ***@***.***>
Sent: Monday, July 25, 2022 11:24 AM
To: IBM/ibmi-bob ***@***.***>
Cc: Susan Bestgen ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [IBM/ibmi-bob] Override COMMIT on SQL table create from default (*CHG) to *NONE not processed (Issue #121)

Hi Susan, what was the contents of RESULTS.TABLE stream file? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: <IBM/ibmi-bob/issues/121/1194313902
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

Hi Susan, what was the contents of RESULTS.TABLE stream file?

**Comment by edmundreinhardt** [comment]
@sbestgen 
> The RESULTS stream files does a CREATE TABLE AS, built over the RSPLOG table. I’ve copied the source for both stream files below. ***@***.*** ***@***.*** 

I don't see the source here.

**Comment by sbestgen** [comment]
Strange.   I did an image copy, let me try text:

CREATE TABLE RESULTS AS
(SELECT REC_SIZE AS REC_SIZE,
        SUM(NUM_READS) AS TOTAL_READS,
        SUM(NUM_WRITES) AS TOTAL_WRITES,
        SUM(NUM_UPDATES) AS TOTAL_UPDATES,
        COUNT(BATCHPF) AS NUM_JOBS,
        AVG(DURATION) AS AVG_TIME,
        SUM( ( CAST( (NUM_READS + NUM_WRITES + NUM_UPDATES) AS FLOAT )
              / DURATION ) * REC_SIZE / 1024 / 1024 ) AS MB_PER_SECOND
FROM RSPLOG
GROUP BY(REC_SIZE)
) WITH DATA


CREATE TABLE RSPLOG
(
BATCHPF CHAR (10) NOT NULL WITH DEFAULT,
TRANSACTIONS BIGINT NOT NULL WITH DEFAULT,
DURATION INT NOT NULL WITH DEFAULT,
NUM_READS INT,
NUM_WRITES INT,
NUM_UPDATES INT,
REC_SIZE INT
)

From: Edmund Reinhardt ***@***.***>
Sent: Monday, July 25, 2022 3:55 PM
To: IBM/ibmi-bob ***@***.***>
Cc: Susan Bestgen ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [IBM/ibmi-bob] Override COMMIT on SQL table create from default (*CHG) to *NONE not processed (Issue #121)

@sbestgen The RESULTS stream files does a CREATE TABLE AS, built over the RSPLOG table. I’ve copied the source for both stream files below. @.*** @.*** I don't see the source here. — Reply to this email directly, view it on GitHub ‍ ‍ ‍
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

@sbestgen<https://github.com/sbestgen>

The RESULTS stream files does a CREATE TABLE AS, built over the RSPLOG table. I’ve copied the source for both stream files below. @.*** @.***

I don't see the source here.

**Comment by edmundreinhardt** [comment]
@sbestgen One thing I noticed is that 
```RESULTS.FILE: $(d)/RESULTS.TABLE```
which should really include a dependency on RSPLOG 
```RESULTS.FILE: $(d)/RESULTS.TABLE RSPLOG.FILE```
Also note that the RSPLOG is not library qualified.  You want to make sure that you have set the objlib attribute because that is what we use as the current schema.

**Comment by edmundreinhardt** [comment]
note that with the latest Bob version you can simplify your Rules.mk to 
`
POOLLOG.FILE: POOLLOG.TABLE
RSPLOG.FILE: RSPLOG.TABLE
SYSLOG.FILE: SYSLOG.TABLE
BATCHPF.FILE: BATCHPF.TABLE
RESULTS.FILE: private COMMIT := *NONE
RESULTS.FILE: RESULTS.TABLE RSPLOG.FILE
`

**Comment by edmundreinhardt** [comment]
@sbestgen is this still a problem?

**Comment by sbestgen** [comment]
@edmundreinhardt Yes, it is still a problem.   I did test what I think should be the fix, which is to add the overridden COMMIT parm in RUNSQLFLAGS:
RUNSQLFLAGS:= DBGVIEW(*SOURCE) TGTRLS($(TGTRLS)) OUTPUT(*PRINT) MARGINS(1024) **COMMIT($(COMMIT))**

I'll create a branch put this change up for review ...

**Comment by edmundreinhardt** [comment]
sounds good

**Comment by sbestgen** [comment]
Fixed with pull request #222
